pkg/roachprod/roachprod.go

```diff
@@ -798,7 +798,7 @@ func updatePrometheusTargets(ctx context.Context, l *logger.Logger, c *install.S
 	if len(nodeIPPorts) > 0 {
 		if err := promhelperclient.NewPromClient().UpdatePrometheusTargets(ctx,
 			envutil.EnvOrDefaultString(prometheusHostUrlEnv, defaultPrometheusHostUrl),
-			c.Name, false, nodeIPPorts, l); err != nil {
+			c.Name, false, nodeIPPorts, !c.Secure, l); err != nil {
```

Comment:
Note that this won't work in the context of the `update-targets` roachprod command: in that case, we'll need to add the `--insecure` option to the command and pass that option when instantiating the cluster (see `startCmd` as an example).